### PR TITLE
Feature/17 Implement Register form

### PR DESCRIPTION
**Pull request by Muzishell**

### CODE REVIEW

File: client/src/pages/Register.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  height: 100%;
+  background-color: #313131;
+`;
+
+const FormContainer = styled.div`
+  margin-top: 20px;
+  height: 100%;
+  width: 100%;
+  display: flex;
+  justify-content: space-evenly;
```
Comment (Takuan-zero) [inline client/src/pages/Register.js]: justify-content: space-around est pas plus sexy ?

Comment (Takuan-zero) [inline client/src/pages/Register.js]: justify-content: space-around est pas plus sexy ?

File: client/package.json [hunk 2]
```diff
@@ -57,7 +58,8 @@
     "webpack": "4.19.1",
     "webpack-dev-server": "3.1.9",
     "webpack-manifest-plugin": "2.0.4",
-    "workbox-webpack-plugin": "3.6.2"
+    "workbox-webpack-plugin": "3.6.2",
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/package.json]: Ce doit être une erreur. J'enlève.

File: client/src/pages/App.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
 
 import LandingPage from 'pages/LandingPage';
+import Register from 'pages/Register';
 
 const App = () => (
   <Router>
-    <Route path="/" component={LandingPage} />
+    <Switch>
+      <Route key={0} exact path="/" component={LandingPage} />
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/pages/App.js]: Removed
911b415

File: client/src/pages/App.js [hunk 4]
```diff
@@ -1,11 +1,15 @@
...
+      <Route key={1} path="/register" component={Register} />
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/pages/App.js]: Done

File: client/src/pages/Register.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import RegisterForm from './register/RegisterForm';
+import Header from './landingPage/Header';
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/pages/Register.js]: Create a directory components/header ?

File: client/src/pages/registerPage/RegisterForm.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    return {
+      email: email || '',
+      password: password || '',
+      username: username || ''
+    }
+  },
+  validationSchema: Yup.object().shape({
+    email: Yup.string().email('Email not valid').required('Email is required'),
+    password: Yup.string().min(9, 'Password must be 9 characters or longer').required('Password is required')
+    // Find a validation for username
```
Comment (agaryen) [inline client/src/pages/registerPage/RegisterForm.js]: I'd remove this comment and add an issue about that.

File: client/src/pages/registerPage/RegisterForm.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  width: 80%;
+  text-align: center;
+`;
+
+const FieldContainer = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const errorDisplay = (touched, errors) => {
```
Comment (agaryen) [inline client/src/pages/registerPage/RegisterForm.js]: Not sure I understand exactly how you handle your errors. Could be nice we discuss this together.

File: client/src/components/FormComponents/Error.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import styled from 'styled-components'
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/components/FormComponents/Error.js]: 642c659

File: client/src/components/FormComponents/Error.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+
+export default styled.p`
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/components/FormComponents/Error.js]: 9ba5ae7

File: client/src/components/FormComponents/Error.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+  color: white;
+  background: red;
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/components/FormComponents/Error.js]: Noted

File: client/src/pages/registerPage/RegisterForm.js [hunk 11]
```diff
@@ -0,0 +1,101 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import { withFormik, Form } from 'formik'
+import styled from 'styled-components'
+import * as Yup from 'yup';
```
[the remark this answers is not in the document]
Comment (Muzishell) [reply client/src/pages/registerPage/RegisterForm.js]: 4adba3f

**Pull request merged**